### PR TITLE
Add Alt/Plus instances for HashMap

### PR DESCRIPTION
**Pull request by alexfmpe**

### CODE REVIEW

File: src/Data/Functor/Plus.hs [hunk 1]
```diff
@@ -120,6 +126,11 @@ instance Plus Seq where
   zero = mempty
 #endif
 
+#ifdef MIN_VERSION_unordered_containers
+instance (Hashable k, Eq k) => Plus (HashMap k) where
```
Comment (RyanGlScott) [inline src/Data/Functor/Plus.hs]: Isn't this `Eq k` constraint unnecessary?

File: src/Data/Functor/Alt.hs [hunk 2]
```diff
@@ -55,7 +55,7 @@ import Data.List.NonEmpty (NonEmpty(..))
 import qualified Data.Monoid as Monoid
 import Data.Semigroup (Option(..), Semigroup(..))
 import qualified Data.Semigroup as Semigroup
-import Prelude (($),Either(..),Maybe(..),const,IO,Ord,(++),(.),either,seq,undefined)
+import Prelude (($),Either(..),Maybe(..),const,IO,Eq,Ord,(++),(.),either,seq,undefined)
```
Comment (RyanGlScott) [inline src/Data/Functor/Alt.hs]: Move this `Eq` import to be within the `#ifdef MIN_VERSION_unordered_containers ... #endif` CPP, as it's only used there.

**Comment by RyanGlScott** [comment]
LGTM, aside from some minor suggestions. Also, don't forget to update the `CHANGELOG`!

File: src/Data/Functor/Plus.hs (hunk 1 above)
Comment (Taneb) [reply src/Data/Functor/Plus.hs]: Remember that `Plus` is a subclass of `Alt`. The `Eq` is necessary for the `Alt` implementation, so is necessary here.

Comment (alexfmpe) [reply src/Data/Functor/Plus.hs]: Afraid not: https://hackage.haskell.org/package/unordered-containers-0.2.10.0/docs/Data-HashMap-Lazy.html#v:union

Comment (RyanGlScott) [reply src/Data/Functor/Plus.hs]: Oops, I forgot about the superclass. Never mind!

**Pull request merged**